core/src/test/java/org/opensearch/sql/expression/json/JsonFunctionsTest.java

```diff
@@ -216,5 +218,153 @@ void json_returnsSemanticCheckException() {
+
+    // null string literal
+    assertEquals(LITERAL_NULL, DSL.jsonExtract(DSL.literal("null"), DSL.literal("$.a")).valueOf());
+
+    // null json
+    assertEquals(
+        LITERAL_NULL, DSL.jsonExtract(DSL.literal(LITERAL_NULL), DSL.literal("$.a")).valueOf());
+
+    // missing json
+    assertEquals(
+        LITERAL_MISSING,
+        DSL.jsonExtract(DSL.literal(LITERAL_MISSING), DSL.literal("$.a")).valueOf());
+
+    // array out of bounds
+    execute_extract_json(LITERAL_NULL, "{\"a\":[1,2,3]}", "$.a[4]");
+  }
+
+  @Test
+  void json_extract_throws_SemanticCheckException() {
+    // invalid path
+    SemanticCheckException invalidPathError =
+        assertThrows(
+            SemanticCheckException.class,
+            () -> DSL.jsonExtract(DSL.literal("{\"a\":1}"), DSL.literal("$a")).valueOf());
+    assertEquals(
+        "JSON path '$a' is not valid. Error details: Illegal character at position 1 expected"
+            + " '.' or '['",
+        invalidPathError.getMessage());
+
+    // invalid json
+    SemanticCheckException invalidJsonError =
+        assertThrows(
+            SemanticCheckException.class,
+            () ->
+                DSL.jsonExtract(
+                        DSL.literal("{\"invalid\":\"json\", \"string\"}"), DSL.literal("$.a"))
+                    .valueOf());
+    assertTrue(
+        invalidJsonError
+            .getMessage()
+            .startsWith(
+                "JSON string '{\"invalid\":\"json\", \"string\"}' is not valid. Error"
+                    + " details:"));
+  }
+
+  @Test
+  void json_extract_throws_ExpressionEvaluationException() {
+    // null path
+    assertThrows(
+        ExpressionEvaluationException.class,
+        () -> DSL.jsonExtract(DSL.literal("{\"a\":1}"), DSL.literal(LITERAL_NULL)).valueOf());
+
+    // missing path
+    assertThrows(
+        ExpressionEvaluationException.class,
+        () -> DSL.jsonExtract(DSL.literal("{\"a\":1}"), DSL.literal(LITERAL_MISSING)).valueOf());
+  }
+
+  @Test
+  void json_extract_search_list_of_paths() {
+    final String objectJson =
+        "{\"foo\": \"foo\", \"fuzz\": true, \"bar\": 1234, \"bar2\": 12.34, \"baz\": null, "
+            + "\"obj\": {\"internal\": \"value\"}, \"arr\": [\"string\", true, null]}";
+
+    ExprValue expected =
+        new ExprCollectionValue(
+            List.of(new ExprStringValue("foo"), new ExprFloatValue(12.34), LITERAL_NULL));
+    Expression pathExpr1 = DSL.literal(ExprValueUtils.stringValue("$.foo"));
+    Expression pathExpr2 = DSL.literal(ExprValueUtils.stringValue("$.bar2"));
+    Expression pathExpr3 = DSL.literal(ExprValueUtils.stringValue("$.potato"));
+    Expression jsonExpr = DSL.literal(ExprValueUtils.stringValue(objectJson));
+    ExprValue actual = DSL.jsonExtract(jsonExpr, pathExpr1, pathExpr2, pathExpr3).valueOf();
+    assertEquals(expected, actual);
+  }
+
+  private static void execute_extract_json(ExprValue expected, String json, String path) {
```

Comment:
Can this be generalized to allow multiple paths, so you can use it for `json_extract_search_list_of_paths` as well?

```suggestion
  private static void execute_extract_json(ExprValue expected, String json, String... paths) {
```